File: SupplierPortal.Web/Modules/ScheduledEmail/Email/EmailForm.cs
using Serenity.ComponentModel;

namespace SupplierPortal.ScheduledEmail.Forms;

[FormScript("ScheduledEmail.Email")]
[BasedOnRow(typeof(EmailRow), CheckNames = true)]
public class EmailForm
{
    public int OfferSupplierId { get; set; }
    public int OfferId { get; set; }
    public bool IsSended { get; set; }
    public bool IsAutoSended { get; set; }
}

File: SupplierPortal.Web/Modules/ServerTypes/ScheduledEmail/EmailForm.ts
import { PrefixedContext, initFormType } from "@serenity-is/corelib";
import { IntegerEditor, BooleanEditor } from "serenity.corelib";

export interface EmailForm {
    OfferSupplierId: IntegerEditor;
    OfferId: IntegerEditor;
    IsSended: BooleanEditor;
    IsAutoSended: BooleanEditor;
}

export class EmailForm extends PrefixedContext {
    static readonly formKey = 'ScheduledEmail.Email';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmailForm.init)  {
            EmailForm.init = true;

            var w0 = IntegerEditor;
            var w1 = BooleanEditor;

            initFormType(EmailForm, [
                'OfferSupplierId', w0,
                'OfferId', w0,
                'IsSended', w1,
                'IsAutoSended', w1
            ]);
        }
    }
}